### PR TITLE
Add course runs endpoint.

### PR DESCRIPTION
**Pull request by peter-fogg**
## [ECOM-3898](https://openedx.atlassian.net/browse/ECOM-3898)

@bderusha @clintonb I've based this PR off Bill's branch. The code will need some updating once the data model is finalized, but this is a start for now.

### CODE REVIEW

**Comment by peter-fogg** [comment]
Note that tests are passing, but the CI build is failing because of quality violations. Since we've changed the interface of `Course`, some old code (which is getting skipped in tests for now) has references to nonexistent class members.

File: course_discovery/apps/api/v1/urls.py [hunk 1]
```diff
@@ -8,5 +8,6 @@
 router = routers.SimpleRouter()
 router.register(r'catalogs', views.CatalogViewSet)
 router.register(r'courses', views.CourseViewSet, base_name='course')
+router.register(r'course-runs', views.CourseRunViewSet, base_name='course-run')
```
Comment (clintonb) [inline course_discovery/apps/api/v1/urls.py]: Silly, I know, but we need to use `course_runs` to support Slumber. See https://github.com/samgiles/slumber/issues/44.

File: course_discovery/apps/api/serializers.py [hunk 2]
```diff
@@ -23,3 +24,19 @@ class ContainedCoursesSerializer(serializers.Serializer):  # pylint: disable=abs
...
+
+class PersonSerializer(serializers.ModelSerializer):
+
+    class Meta(object):
+        fields = ('id', 'key')
+        model = Person
+
+
+class CourseRunSerializer(serializers.ModelSerializer):
+    url = serializers.HyperlinkedIdentityField(view_name='api:v1:course-run-detail', lookup_field='key')
```
Comment (clintonb) [inline course_discovery/apps/api/serializers.py]: No URLs for now.

File: course_discovery/apps/api/v1/views.py [hunk 3]
```diff
@@ -143,3 +149,13 @@ def list(self, request, *args, **kwargs):  # pylint: disable=unused-argument
     def retrieve(self, request, *args, **kwargs):
         """ Retrieve details for a course. """
         return super(CourseViewSet, self).retrieve(request, *args, **kwargs)
+
+
+class CourseRunViewSet(viewsets.ReadOnlyModelViewSet):
+    """ Course run resource."""
+    lookup_field = 'key'
+    lookup_value_regex = COURSE_ID_REGEX
+    permission_classes = (IsAuthenticated,)
```
Comment (clintonb) [inline course_discovery/apps/api/v1/views.py]: This and `pagination_class` are [defined in settings](https://github.com/edx/course-discovery/blob/master/course_discovery/settings/base.py#L236-L259). No need to re-define them.

File: course_discovery/apps/api/serializers.py [hunk 4]
```diff
@@ -23,3 +24,19 @@ class ContainedCoursesSerializer(serializers.Serializer):  # pylint: disable=abs
...
+    people = PersonSerializer(many=True)
```
Comment (bderusha) [inline course_discovery/apps/api/serializers.py]: The spec exposes this as `staff`

File: course_discovery/apps/api/v1/urls.py (hunk 1 above)
Comment (peter-fogg) [inline course_discovery/apps/api/v1/urls.py]: Blergh. Fixed.

File: course_discovery/apps/api/serializers.py (hunk 2 above)
Comment (peter-fogg) [inline course_discovery/apps/api/serializers.py]: Fixed.

File: course_discovery/apps/api/v1/views.py (hunk 3 above)
Comment (peter-fogg) [inline course_discovery/apps/api/v1/views.py]: Didn't notice that; fixed.

File: course_discovery/apps/api/serializers.py (hunk 4 above)
Comment (peter-fogg) [inline course_discovery/apps/api/serializers.py]: Updated.

**Pull request closed without merging**